src/QsCompiler/Tests.Compiler/TestUtils/Signatures.fs

```diff
@@ -609,6 +621,195 @@ let public ClassicalControlSignatures =
+                                                                      OpProperty.Controllable ]
+                         InferredInformation = InferredCallableInformation.NoInformation
+                     })
+                    |> QsTypeKind.Operation |]
+
+/// Expected callable signatures to be found when running Lambda Lifting tests
+let public LambdaLiftingSignatures =
+    [| // Basic Lift
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [||], "Int" // The generated operation
+         |])
+        // Without Return Value
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [||], "Unit" // The generated operation
+         |])
+        // Call Valued Callable
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "Bar", [||], "Int"
+             LambdaLiftingNS, "_Foo", [||], "Int" // The generated operation
+         |])
+        // Call Unit Callable
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "Bar", [||], "Unit"
+             LambdaLiftingNS, "_Foo", [||], "Unit" // The generated operation
+         |])
+        // Call Valued Callable Recursive
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Int" // The original operation
+             LambdaLiftingNS, "_Foo", [||], "Int" // The generated operation
+         |])
+        // Call Unit Callable Recursive
+        (_DefaultTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [||], "Unit" // The generated operation
+         |])
+        // Use Closure
+        (_WithTupleTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [| "Double"; "String"; "Result"; "Unit" |], "(Double, String, Result)" // The generated operation
+         |])
+        // Use Lots of Params
+        (_WithTupleTypes,
+         [|
+             LambdaLiftingNS, "Foo", [||], "Unit" // The original operation
+             LambdaLiftingNS, "_Foo", [| "Int" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int"; "Double" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int"; "Double"; "String" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int"; "(Double, String)" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int"; "Double"; "String" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "(Int, Double)"; "String" |], "Unit"
+             LambdaLiftingNS, "_Foo", [| "Int"; "Double"; "String" |], "Unit"
```

Comment:
I don't think I understand the naming here. The lifted lambdas all have the same name `_Foo`?

Reply:
They all end with "_Foo". Since I can't know the generated names of the lifted operations here, this field is for how the name of the operation ends.

Reply:
> Since I can't know the generated names of the lifted operations here

You could use a deterministic naming scheme, like numbering them by the order they appear. That would make it easier to write tests with asserts for specific lambdas.

Reply:
I'm not interested in changing the naming scheme that's used throughout the compiler. I'm not sure there would be a significant benefit to that, and it would take a bit of looking into to make it work right.

Reply:
In terms of benefits from testing, I might be missing something but I'm not sure how you are testing that the lambdas are substituted in the right places. For example, how do you assert that:

```qsharp
F(x -> x + 1, x -> x + 2);
```

transforms into:

```qsharp
function Lambda1(x : Int) : Int {
    return x + 1;
}

function Lambda2(x : Int) : Int {
    return x + 2;
}

F(Lambda1, Lambda2);
```

and not (incorrectly):

```qsharp
F(Lambda2, Lambda1);
```

Would this be caught by the existing test suite?

Reply:
> There are several tests so far that check that the original location of the lambda is replaced with the call to the lifted operation.

Yes, I saw the tests that check that the lambda is replaced by a reference to a lifted lambda, but it didn't look like they distinguish between which lifted lambda, if there is more than one. If the testing infrastructure can distinguish between the two cases above that would be good to know.

Reply:
So in terms of the test infrastructure, I think I'm looking for a test case where you have something like

```fsharp
GetCallablesWithSuffix result Signatures.LambdaLiftingNS "_Foo"
```

but *without* expecting there to be one element in the returned sequence. Right now all the tests that use `GetCallablesWithSuffix` also call `Seq.exactlyOne`. Then the test would determine which lambda has which body, and assert that the original lambdas were replaced with the correct corresponding lifted ones.

Reply:
I was thinking of the `identifyGeneratedByCalls` function, but that is limited to looking at lines that are just call expressions. I've added the tests case you suggested above, and just used logic written in that test case. If it is something I use more as I continue to refactor code, I will look into making that logic a TestUtil function as well.